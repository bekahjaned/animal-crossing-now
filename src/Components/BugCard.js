import React from "react";

class BugCard extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="critter">
        <img
          src={require(`../data/images/Bugs/${this.props.image}.png`)}
          alt="bug"
        ></img>
        <p className="critter-name">{this.props.name}</p>
        <p>{this.props.location}</p>
      </div>
    );
  }
}

export default BugCard;
