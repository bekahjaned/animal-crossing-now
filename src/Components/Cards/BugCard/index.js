import React from "react";
import "../index.css";

class BugCard extends React.Component {
  render() {
    return (
      <div className="critter">
        <img
          src={require(`../../../data/images/Bugs/${this.props.image}.png`)}
          alt="bug"
        ></img>
        <p className="critter-name">{this.props.name}</p>
        <p>{this.props.location}</p>
      </div>
    );
  }
}

export default BugCard;
