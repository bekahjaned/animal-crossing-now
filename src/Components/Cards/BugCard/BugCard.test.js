import React from "react";
import renderer from "react-test-renderer";

import BugCard from "./index";

it("renders a BugCard", () => {
  // given
  const image = "Cricket";
  const name = "Cricket";
  const location = "On ground";
  const availability = "17 - 8";
  const price = "130";

  // when
  const tree = renderer
    .create(
      <BugCard
        image={image}
        name={name}
        location={location}
        availability={availability}
        price={price}
      />
    )
    .toJSON();

  //   then
  expect(tree).toMatchSnapshot();
});
