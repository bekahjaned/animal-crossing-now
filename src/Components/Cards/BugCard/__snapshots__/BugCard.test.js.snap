// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a BugCard 1`] = `
<div
  className="critter"
>
  <img
    alt="bug"
    src="Cricket.png"
  />
  <p
    className="critter-name"
  >
    Cricket
  </p>
  <p>
    On ground
  </p>
  <p>
    Available: 
    17 - 8
  </p>
</div>
`;
